File: src/FrameIO/Models/ComPortConfigModels.cs
using FrameIO.Data;
using System;
using static FrameHelper.EntityHelpers;

namespace FrameIO.Models
{
    public interface IComPortConfigDTO : IWithIdAndTimeStamp
    {
        int Number { get; set; }
        string PortName { get; set; }
        Guid ComPortTypeId { get; set; }
        Guid ComDeviceConfigId { get; set; }
    }

    public class ComPortConfigDTO : IComPortConfigDTO
    {
        public Guid Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public int Number { get; set; }
        public string PortName { get; set; }
        public Guid ComPortTypeId { get; set; }
        public Guid ComDeviceConfigId { get; set; }
    }

    public interface IComPortConfigView : IComPortConfigDTO
    {
    }

    public class ComPortConfigView : ComPortConfigDTO, IComPortConfigView
    {
    }

}

File: src/FrameUI/src/app/models/ComPortConfigModels.ts


export interface IcomPortConfigDTO {
    id: string;
    timeStamp: Date;
    number: number;
    portName: string;
    comPortTypeId: string;
    comDeviceConfigId: string;
    
}

export class ComPortConfigDTO implements IcomPortConfigDTO {
    public id: string;
    public timeStamp: Date;
    public number: number;
    public portName: string;
    public comPortTypeId: string;
    public comDeviceConfigId: string;
    
    constructor(model?: IcomPortConfigDTO) {
        
        if(model) {
            this.id = model.id;
            this.timeStamp = model.timeStamp;
            this.number = model.number;
            this.portName = model.portName;
            this.comPortTypeId = model.comPortTypeId;
            this.comDeviceConfigId = model.comDeviceConfigId;
            
        }
    }
}

export interface IcomPortConfigView extends IcomPortConfigDTO {
    
}

export class ComPortConfigView extends ComPortConfigDTO implements IcomPortConfigView {
    
    constructor(model?: IcomPortConfigView) {
        super(model);
        if(model) {
            
        }
    }
}
 